common/model/bundle.c

```diff
@@ -275,23 +303,39 @@ void bundle_reset_indexes(bundle_transactions_t *const bundle) {
   }
 }
 
-void bundle_set_messages(bundle_transactions_t *bundle, signature_fragments_t *messages) {
-  iota_transaction_t *tx = NULL;
-  size_t index = 0;
-  tryte_t **msg = NULL;
-  size_t msg_len = 0;
-
-  BUNDLE_FOREACH(bundle, tx) {
-    msg = signature_fragments_at(messages, index);
-    if (msg == NULL) {
-      break;
+void bundle_set_message_string(bundle_transactions_t *bundle, char const *const msg) {
+  // converting ascii string to trytes string
+  size_t trytes_msg_len = strlen(msg) * 2;
+  tryte_t *trytes_msg = malloc(sizeof(tryte_t) * (trytes_msg_len + 1));
+  ascii_to_trytes(msg, trytes_msg);
+  trytes_msg[trytes_msg_len] = '\0';  // null terminator is needed.
+
+  // apply trytes string to bundle
+  iota_transaction_t tx = {};
+  tryte_t *msg_buff = malloc(sizeof(tryte_t) * (NUM_TRYTES_MESSAGE + 1));
+  size_t msg_chunks = floor((double)trytes_msg_len / NUM_TRYTES_MESSAGE) + 1;
+  signature_fragments_t *sign_fragments = signature_fragments_new();
+  size_t last_chunk = 0;
+  transaction_reset(&tx);
+
+  for (size_t i = 0; i < (msg_chunks * NUM_TRYTES_MESSAGE); i += NUM_TRYTES_MESSAGE) {
+    if (i + NUM_TRYTES_MESSAGE > trytes_msg_len) {
+      last_chunk = trytes_msg_len - ((msg_chunks - 1) * NUM_TRYTES_MESSAGE);
+      strncpy((char *)msg_buff, (char *)trytes_msg + i, last_chunk);
```

Comment:
I see there are lot of string length counting to prevent from `strncpy` result doesn't end with null character. Maybe we can switch to `snprintf` which does not write more bytes allocated by string of destination?

Reply:
`snprintf` is slow. it's not good for chunking.